tiling.js

```diff
@@ -4546,6 +4556,26 @@ export function setFocusMode(mode, space) {
     const workArea = space.workArea();
     const selectedWin = space.selectedWindow;
     // if centre also center selectedWindow
+    switch (mode) {
+    case FocusModes.CENTER:
+        if (selectedWin) {
+            // check it closer to min or max of workArea
+            const frame = selectedWin.get_frame_rect();
+            const winMidpoint = space.visibleX(selectedWin) + frame.width / 2;
+            const workAreaMidpoint = workArea.width / 2;
+            if (winMidpoint <= workAreaMidpoint) {
+                space.unfocusXPosition = 0;
+            } else {
+                space.unfocusXPosition = workArea.width;
+            }
+            centerWindowHorizontally(selectedWin);
+        }
+        break;
+    default:
+        // for other modes run a `layout` call to action the mode
```

Comment:
oh, it was that simple to not have to repeat code there

Reply:
:+1: I've pushed another one for some more cleanup (forgot to remove manual if check which the above code replaced).